backend/src/sprint/sprint.controller.ts

```diff
@@ -109,12 +119,25 @@ export class SprintController {
   @Post(':sprintId/add-story/:storyId')
   async addStoryToSprint(@Token() token: TokenDto, @Param('sprintId', ParseIntPipe) sprintId: number, @Param('storyId', ParseIntPipe) storyId: number) {
     const story = await this.storyService.getStoryById(storyId);
-    if (story == null)
+    if (!story)
       throw new BadRequestException('No story with the given ID exists');
 
+    if (story.timeComplexity < 1)
+      throw new BadRequestException('Time complexity of a story cannot be negative or zero.');
+
+    if (story.isRealized)
+      throw new BadRequestException('The story is already realized.');
+
+    if (await this.storyService.isStoryInActiveSprint(storyId))
+      throw new BadRequestException('The story is already in an active sprint.');
+
     const sprint = await this.sprintService.getSprintById(sprintId);
-    if (sprint == null)
-      throw new BadRequestException('No sprint with the given ID exists in the ')
+    if (!sprint)
+      throw new BadRequestException('No sprint with the given ID exists in the ');
+
+    const storyIdsInSprint = (await this.storyService.getStoryIdsForSprint(storyId)).length;
```

Comment:
Tale check NI OK. Sprint velocity je v točkah, in to ni število storyev, kvečjemu bi lahko dal sprint.velocity < vsota uteži storyev